src/Tasks/CreateManifestResourceName.cs

```diff
@@ -161,8 +163,19 @@ CreateFileStream createFileStream
                     // If opted into convention and no DependentUpon metadata and is a resx file, reference "<filename>.cs" if it exists.
                     if (isResxFile && UseDependentUponConvention && string.IsNullOrEmpty(dependentUpon))
                     {
+                        // Assume that by convention the expected file name is "<filename>.cs"
                         string conventionDependentUpon = Path.ChangeExtension(Path.GetFileName(fileName), SourceFileExtension);
 
+                        // Verify that the file name didn't have a culture associated with it. Ex: "<filename>.<culture>.resx" If we don't strip the culture we look for TestComponent.de.cs, which we don't want.
+                        if (resourceFile.GetMetadata("WithCulture") == "true")
```

Comment:
How confident are we that this is always set for culture-specific resx?

Reply:
I didn't consider the possibility that WithCulture could be empty while there is some Culture. I'll modify this to instead check if Culture has a value.

Reply:
After some digging: [SplitResourcesByCulture](https://github.com/microsoft/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets#L2924) sets `WithCulture` as "true" or "false" depending on the file name. We already confirmed that `SplitResourcesByCulture` runs before `CreateManifestResourceName`, so no actual code change needed here.